src/quo2/components/wallet/network_routing/view.cljs

```diff
@@ -0,0 +1,197 @@
+           :accessibility-label :network-routing-bar}
+          [gesture/gesture-detector
+           {:gesture
+            (-> (gesture/gesture-pan)
+                (gesture/enabled @detecting-gesture?)
+                (gesture/on-begin
+                 (fn [_]
+                   (animation/increase-slider slider-width-shared-value slider-height-shared-value)
+                   (reset! amount-on-gesture-start (reanimated/get-shared-value amount-shared-value))))
+                (gesture/on-update
+                 (fn [event]
+                   (let [new-amount (-> (oops/oget event "translationX")
+                                        (width->amount)
+                                        (+ @amount-on-gesture-start)
+                                        (utils.number/value-in-range 1 max-amount))]
+                     (reanimated/set-shared-value amount-shared-value new-amount))))
+                (gesture/on-finalize
+                 (fn [_]
+                   (animation/decrease-slider slider-width-shared-value slider-height-shared-value)
+                   (animation/hide-slider slider-opacity-shared-value)
+                   (on-new-amount (reanimated/get-shared-value amount-shared-value))
+                   (reset! detecting-gesture? false))))}
+           [:f> slider
+            {:width-shared-value   slider-width-shared-value
+             :height-shared-value  slider-height-shared-value
+             :opacity-shared-value slider-opacity-shared-value}]]]]))))
+
+(defn- add-bar-shared-values
+  [{:keys [amount] :as network}]
+  (assoc network
+         :amount-shared-value      (reanimated/use-shared-value amount)
+         :translate-x-shared-value (reanimated/use-shared-value 0)))
+
+(def ^:private get-negative-amount
+  (comp - reanimated/get-shared-value :amount-shared-value))
+
+(defn network-routing-bars
+  [_]
+  (let [selected-network-idx (reagent/atom nil)
+        selecting-network?   (reagent/atom false)
+        press-locked?        (reagent/atom false)
+        lock-press           #(reset! press-locked? true)
+        unlock-press         #(reset! press-locked? false)
+        hide-bar-animations  (atom {})
+        add-hide-animation   (fn [k anim-props]
+                               (let [new-anim #(animation/hide-pressed-bar anim-props)]
+                                 (swap! hide-bar-animations assoc k new-anim)))
+        reset-state-values   (fn []
+                               (reset! selected-network-idx nil)
+                               (reset! selecting-network? false))]
+    (fn [{:keys [networks total-width total-amount requesting-data? on-amount-selected
+                 add-new-timeout-fn]}]
+      (let [bar-opacity-shared-value (reanimated/use-shared-value 0)
+            network-bars             (map add-bar-shared-values networks)
+            amount->width            #(* % (/ total-width total-amount))
```

Comment:
`total-amount` can be zero if we pass `0` to the component. Does it make sense to mount/use this component to select network amounts when the total amount is zero? I think no :thinking:  so I'd say just adding that precondition to this component is enough.

Reply:
> Does it make sense to mount/use this component to select network amounts when the total amount is zero?

That's exactly my point of view, the component shouldn't assume how it's going to be used too much. Maybe in the future the component will change a bit, and this assumption about when it should render won't hold anymore, but devs more likely won't update the code to avoid the division by zero and calculations with infinity.

It's just a more fragile implementation @ulisesmac, but it's okay for now :)